tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -7517,6 +7524,103 @@ Result D3D12Device::RayTracingPipelineStateImpl::createShaderTables(
 
 #endif // SLANG_GFX_HAS_DXR_SUPPORT
 
+Result createNullDescriptor(
+    ID3D12Device* d3dDevice,
+    D3D12_CPU_DESCRIPTOR_HANDLE destDescriptor,
+    const D3D12Device::ShaderObjectLayoutImpl::BindingRangeInfo& bindingRange)
+{
+    switch (bindingRange.bindingType)
+    {
+    case slang::BindingType::ConstantBuffer:
+        {
+            D3D12_CONSTANT_BUFFER_VIEW_DESC cbvDesc = {};
+            cbvDesc.BufferLocation = 0;
+            cbvDesc.SizeInBytes = 0;
+            d3dDevice->CreateConstantBufferView(&cbvDesc, destDescriptor);
+        }
+        break;
+    case slang::BindingType::MutableRawBuffer:
+        {
+            D3D12_UNORDERED_ACCESS_VIEW_DESC uavDesc = {};
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_BUFFER;
+            uavDesc.Buffer.Flags = D3D12_BUFFER_UAV_FLAG_RAW;
+            uavDesc.Format = DXGI_FORMAT_R32_TYPELESS;
+            d3dDevice->CreateUnorderedAccessView(nullptr, nullptr, &uavDesc, destDescriptor);
+        }
+        break;
+    case slang::BindingType::MutableTypedBuffer:
+        {
+            D3D12_UNORDERED_ACCESS_VIEW_DESC uavDesc = {};
+            uavDesc.ViewDimension = D3D12_UAV_DIMENSION_BUFFER;
+            uavDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+            d3dDevice->CreateUnorderedAccessView(nullptr, nullptr, &uavDesc, destDescriptor);
+        }
+        break;
+    case slang::BindingType::RawBuffer:
+        {
+            D3D12_SHADER_RESOURCE_VIEW_DESC srvDesc = {};
+            srvDesc.ViewDimension = D3D12_SRV_DIMENSION_BUFFER;
+            srvDesc.Buffer.Flags = D3D12_BUFFER_SRV_FLAG_RAW;
+            srvDesc.Format = DXGI_FORMAT_R32_TYPELESS;
+            srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+            d3dDevice->CreateShaderResourceView(nullptr, &srvDesc, destDescriptor);
+        }
+        break;
+    case slang::BindingType::TypedBuffer:
+        {
+            D3D12_SHADER_RESOURCE_VIEW_DESC srvDesc = {};
+            srvDesc.ViewDimension = D3D12_SRV_DIMENSION_BUFFER;
+            srvDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+            srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+            d3dDevice->CreateShaderResourceView(nullptr, &srvDesc, destDescriptor);
+        }
+        break;
+    case slang::BindingType::Texture:
+        {
+            D3D12_SHADER_RESOURCE_VIEW_DESC srvDesc = {};
+            srvDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+            srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+            switch (bindingRange.resourceShape)
```

Comment:
I see. This is interesting, because in most cases D3D doesn't care about the difference between various SRV bindings, but obviously this matters for the case of a null binding.

I wonder if we can still avoid storing the `resourceShape` on the binding range...